pconduit/in_memory.go

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pconduit
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/conduitio/conduit-commons/schema"
+)
+
+type inMemoryService struct {
+	// schemas is a map of schema subjects to all the versions of that schema
+	// versioning starts at 1, newer versions are appended to the end of the versions slice.
+	schemas map[string][]schema.Schema
+	// m guards access to schemas
+	m sync.Mutex
+}
+
+func NewInMemoryService() SchemaService {
```

Comment:
Hmm, I don't think the protocol is the right place for this. The purpose of this repo is to provide a thin layer around the protobuf files, while this part adds some actual logic. Granted, simple logic, but still, probably not the right place for this. Given that it will only be used by the connector SDK I'd move it there.

---

Sidenote: in the long run, my proposal would be to aim for something like this:
- implement a full-fledged schema service in a separate repo (e.g. `github.com/conduitio/schema-service`)
  - at first we'll probably do it directly in Conduit, but I think it would be worth extracting at some point, it can be a self contained piece of software
- the schema service should then provide different storage mechanisms, including an "in-memory" one
- we change this in-memory service to use the full-fledged schema service under the hood, so we're sure it works like the real one, because it is a real one